homeassistant/components/shelly/__init__.py

```diff
@@ -479,10 +492,41 @@ def __init__(
         self.entry = entry
         self.device = device
 
+        entry.async_on_unload(
+            self.async_add_listener(self._async_device_updates_handler)
+        )
+        self._last_event: dict[str, Any] | None = None
+
         entry.async_on_unload(
             hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, self._handle_ha_stop)
         )
 
+    @callback
+    def _async_device_updates_handler(self) -> None:
+        """Handle device updates."""
+        if (
+            not self.device.initialized
+            or not self.device.event
+            or self.device.event == self._last_event
+        ):
+            return
+
+        self._last_event = self.device.event
+
+        for event in self.device.event["events"]:
+            if event.get("event") not in RPC_INPUTS_EVENTS_TYPES:
+                continue
+
+            self.hass.bus.async_fire(
+                EVENT_SHELLY_CLICK,
+                {
```

Comment:
Let's include the device type (rpc/block) or generation (1/2). That way people will be able to use blueprints that handle both.

Reply:
Added `ATTR_GENERATION: Final = "generation"` with values of 1 or 2 to the `shelly.click` event